packages/yuudachi/src/functions/reports/resolveReports.ts

```diff
@@ -0,0 +1,51 @@
+import { kSQL, logger } from "@yuudachi/framework";
+import type { Guild, User } from "discord.js";
+import type { Sql } from "postgres";
+import { container } from "tsyringe";
+import { upsertReportLog } from "../logging/upsertReportLog.js";
+import { ReportStatus } from "./createReport.js";
+import type { RawReport } from "./transformReport.js";
+import { updateReport } from "./updateReport.js";
+
+export async function resolvePendingReports(guild: Guild, targetId: string, caseId: number, moderator: User) {
+	const sql = container.resolve<Sql<any>>(kSQL);
+
+	const pendingReports = await sql<RawReport[]>`
+		select *
+		from reports
+		where guild_id = ${guild.id}
+			and status = ${ReportStatus.Pending}
+			and (target_id = ${targetId} or author_id = ${targetId})
```

Comment:
Should we resolve reports (as Spam) where the actioned user is the author? Or only the one where it is the target?

Reply:
I don't think so? The actioned user could make a legitimate report, but still be reported/actioned themselves, no?

Reply:
Yeah, thats what I thought too. Give me a sec to remove